lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -555,6 +511,13 @@ static const llvm::StringMap<AnnoRecord> annotationRecords{{
     {wiringSourceAnnoClass, {stdResolve, applyWiring}},
     {attributeAnnoClass, {stdResolve, applyAttributeAnnotation}}}};
 
+LogicalResult registerAnnotationRecord(StringRef annoClass,
+                                       AnnoRecord annoRecord) {
+  return LogicalResult::success(
+      annotationRecords.insert({annoClass, annoRecord}).second);
```

Comment:
It would be cleaner to have this produce a standard error message here as opposed to relying on the user registering the annotation to create the error message.